core/group/service.go

```diff
@@ -369,5 +369,12 @@ func (s Service) Delete(ctx context.Context, id string) error {
 		return err
 	}
 
+	group, err := s.repository.GetByID(ctx, id)
```

Comment:
better we fetch it before relation service? So if in case it doesn't exist, it fail early?

Reply:
Changed the logic to fetch the group early, and emit the event only after both, relation as well as repository rows have been deleted.